Makefile

```diff
@@ -10,63 +10,71 @@ ES_IMAGE_TAG=docker.io/library/elasticsearch:6.8.12
 LOKI_CONTAINER_NAME=loki
 LOKI_IMAGE_TAG=docker.io/grafana/loki:2.2.1
 
-all: clean build test build-image local
+##@ <target>:
+all: ## Run everything (clean, build, test...)
+	clean build test build-image local
 
 lint: $(GOLINT)
 	 $(GOLINT) main.go
 
-build: lint
+build: ## Build the executable
+	make lint
 	go build -o logger -v main.go
 
-test: lint
+test: ## Run the tests
+	make lint
 	go test -v loadclient/*.go
 
-clean:
+clean: ## Delete the object files and cached files including the executable "logger"
 	rm -f ./logger
 	go clean ./...
 
-build-image:
+build-image: ## Build the image
 	docker build -t $(IMAGE_PREFIX)/cluster-logging-load-client .
 	docker tag $(IMAGE_PREFIX)/cluster-logging-load-client $(IMAGE_PREFIX)/cluster-logging-load-client:$(IMAGE_TAG)
 
-push-image:
+push-image: ## Push the image
 	docker push $(IMAGE_PREFIX)/cluster-logging-load-client:$(IMAGE_TAG)
 	docker push $(IMAGE_PREFIX)/cluster-logging-load-client:latest
 
-deploy:
+deploy: ## Deploy the image (build-image must be called before deploy)
 	kubectl apply -f deployment.yaml
 
-undeploy:
+help: ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+undeploy: ## Undeploy the image
 	kubectl delete -f deployment.yaml
 
-local: clean-local deploy-local-es deploy-local-loki run-local-es-generate run-local-es-query run-local-loki-generate run-local-loki-query
+local: ## Run all the local commands
```

Comment:
you need to move all dependent targets to the same line :-( ... please fix in all places 

local:  clean-local deploy-local-es deploy-local-loki run-local-es-generate run-local-es-query run-local-loki-generate run-local-loki-query ## Run all the local commands